File: project_euler/Problem067/Problem067.py
class Problem067:
    def solution(self) -> None:
        file_name, border = 'triangle.txt', 100
        target = [[0 for _ in range(border)] for _ in range(border)]
        with open(file_name, 'r') as handle:
            index = 0
            for line in handle:
                elements = line.strip().split(' ')
                for i, element in enumerate(elements):
                    target[index][i] = int(element)
                index += 1

        for i in range(border - 1, 0, -1):
            for j in range(i - 1, -1, -1):
                target[i - 1][j] += max(target[i][j], target[i][j + 1])
        print(target[0][0])

File: project_euler/Problem067/Problem067.go
package project_euler

import (
    "bufio"
    "fmt"
    "io"
    "math"
    "os"
    "strconv"
    "strings"
)

func Problem067() {
    fileName := "triangle.txt"
    border := 100
    target := make([][]int, border)
    for i := 0; i < border; i++ {
        target[i] = make([]int, border)
    }

    file, _ := os.Open(fileName)
    handle := bufio.NewReader(file)
    index := 0
    for true {
        line, _, err := handle.ReadLine()
        if io.EOF == err {
            break
        }

        elements := strings.Split(strings.Replace(string(line), "\n", "", -1), " ")
        for i, element := range elements {
            auxiliary, _ := strconv.Atoi(element)
            target[index][i] = auxiliary
        }
        index++
    }
    file.Close()

    for i := border - 1; i > 0; i-- {
        for j := i - 1; j >= 0; j-- {
            target[i - 1][j] += int(math.Max(float64(target[i][j]), float64(target[i][j + 1])))
        }
    }
    fmt.Println(target[0][0])
}
